.github/workflows/dev.yml

```diff
@@ -38,7 +37,6 @@ jobs:
       - name: Install and configure Poetry
         uses: snok/install-poetry@v1
         with:
-          version: ${{ matrix.poetry-version }}
```

Comment:
yeah so your now using version 2 of poetry which is not yet supported

Reply:
Please dont change the tests

Or at least do it in an isolated pr without changes your trying to do to the library ahaha

Reply:
i'd meant to revert the tests. reverted in c90a886afc45345b77b45dbf54c55a57a33f5c1b

Reply:
it not working on version 2 of poetry is good information because I've been trying to get it to work with the latest poetry.

my default workflow for installing any python package is to do it in a fresh venv like this:

```
uv venv .venv -p 3.11
source .venv/bin/activate
uv pip install poetry pip setuptools (these are hidden dependencies you have)
make install
```

this uses poetry 2.0.1
```
uv pip list | grep poetry  
poetry             2.0.1
poetry-core        2.0.1
```

Reply:
Gotcha, So we need to bump to poetry v2. 
Lets do that as new pr, and a major version change.
Theres a few templates which use specific version of poetry so they will need to be regenerated as well.